cli/run_util.go

```diff
@@ -849,30 +850,34 @@ func activateAndRunSteps(
 }
 
 func logStepStarted(stepInfo stepmanModels.StepInfoModel, step stepmanModels.StepModel, idx int, stepExcutionId string, stepStartTime time.Time) {
-	idVersion := ""
-	if stepInfo.Step.Title != nil && *stepInfo.Step.Title != "" {
-		idVersion = *stepInfo.Step.Title
-	}
-
-	title := ""
-	if step.Title != nil {
-		title = *step.Title
-	}
-
 	params := log.StepStartedParams{
 		ExecutionId: stepExcutionId,
 		Position:    idx,
-		IdVersion:   idVersion,
+		Title:       stepTitle(stepInfo, step),
 		Id:          stepInfo.ID,
 		Version:     stepInfo.Version,
-		Title:       title,
 		Collection:  stepInfo.Library,
 		Toolkit:     toolkits.ToolkitForStep(step).ToolkitName(),
 		StartTime:   stepStartTime.Format(time.RFC3339),
 	}
 	log.PrintStepStartedEvent(params)
 }
 
+func stepTitle(stepInfo stepmanModels.StepInfoModel, step stepmanModels.StepModel) string {
+	title := ""
+	if stepInfo.Step.Title != nil && *stepInfo.Step.Title != "" {
+		// At this point the title is either the overridden title from the bitrise.yml file or the composite step id.
+		title = *stepInfo.Step.Title
+	}
+
+	// The composite step id is not useful for the user. So let's replace it with the original step title defined in the step.yml file.
+	if strings.HasPrefix(title, stepInfo.ID) && step.Title != nil && *step.Title != "" {
```

Comment:
is it working if the step ID composite has the collection part too?
- `git::<repo_utrl>`
- `https://github.com/bitrise-io/bitrise-steplib.git::script@1`

Reply:
Nice catch. Pushed a new commit with the updated logic.